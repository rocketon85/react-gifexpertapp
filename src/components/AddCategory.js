import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handlerChange = (e) => {
        setInputValue(e.target.value);
    }

    const handlerSubmit = (e) => {
        e.preventDefault();

        if (setCategories(inputValue)) { //setCategories(cats => [...cats, inputValue] );
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handlerSubmit}>
            <p> {inputValue} </p>
            <input 
                type="text"
                value={inputValue}
                onChange={ handlerChange }
            />
        </form>
    )
}

AddCategory.propTypes  = {
    setCategories: PropTypes.func.isRequired
}