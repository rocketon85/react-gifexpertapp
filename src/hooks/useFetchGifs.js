import { useState, useEffect } from 'react';

import { getGif } from '../helpers/getGifs';

export const useFetchGifs = (cat) => {

    const [gifs, setGifs] = useState({
        data:[],
        loading: true
    });

    useEffect( () =>{
        getGif(cat).then(gifs => setGifs({
            data:gifs,
            loading:false
        } ));
    }, [cat])

    return gifs;
}

