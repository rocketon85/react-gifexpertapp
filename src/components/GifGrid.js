import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';


export const GifGrid = ({cat}) => {
   
    const {data:images, loading} = useFetchGifs(cat);
  

    return (
        <>
            <h3>{cat}</h3>

            {loading && <p>Loading</p>}

            <div className='card-grid animate__animated animate__bounce'>
                
                {
                    images.map( img => (
                        <GifGridItem 
                            key={img.id}
                            {...img} 
                        />))
                }
            
            </div>
        </>
    )
}


GifGrid.propTypes  = {
    cat: PropTypes.string.isRequired
}