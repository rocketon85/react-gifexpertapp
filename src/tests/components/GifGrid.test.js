import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('test of GifGrid', () => {
    const category = "Dragon Ball";

  test('should match with snapshot', () => {
    useFetchGifs.mockReturnValue({
        data:[],
        loading: true
    });

    const wrapper = shallow( <GifGrid cat={ category } /> );

    expect( wrapper ).toMatchSnapshot();
    
  });
  
  test('should render with data', () => {
      const imgs = [{
          id: 1,
          title: 'test',
          url: 'http://'
      }]

    useFetchGifs.mockReturnValue({
        data:imgs,
        loading: false
    });

    const wrapper = shallow( <GifGrid cat={ category } /> );

    expect( wrapper ).toMatchSnapshot();

    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(imgs.length);

  });
  
});
