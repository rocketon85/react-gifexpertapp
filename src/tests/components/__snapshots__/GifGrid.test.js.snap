// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test of GifGrid should match with snapshot 1`] = `
<Fragment>
  <h3>
    Dragon Ball
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid animate__animated animate__bounce"
  />
</Fragment>
`;

exports[`test of GifGrid should render with data 1`] = `
<Fragment>
  <h3>
    Dragon Ball
  </h3>
  <div
    className="card-grid animate__animated animate__bounce"
  >
    <GifGridItem
      id={1}
      key="1"
      title="test"
      url="http://"
    />
  </div>
</Fragment>
`;
